src/test/server/mock_socket.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <filesystem>
+#include <fstream>
+
+namespace opossum {
+
+static constexpr char file_name[] = "socket_file";
```

Comment:
The read and write operations within the buffer require a file descriptor. Those methods are wrappers for the OS functions `read` and `write`. Since, I don't want to change the buffer implementation for testing I decided to use files instead of network sockets. My solution for parallel testing would be adding a uuid to the file name.

Reply:
Two options:
1. See the use of tmpname in abstract_visualizer.hpp (and how we delete the file and the string)
2. Opossum::test_data_path

I think the second one is easier and more consistent with other tests

Reply:
Second option seems reasonable. Does the file still need to have a unique name in this case (such as UUID in the file name)?

Reply:
You may assume that there is only one test running in that folder, so unless you are using multiple mock sockets at the same time, you should be good.

Reply:
Good. I only need one mocked socket per test. Gonna remove this UUID stuff...